packages/twenty-front/src/modules/object-record/record-table/hooks/useCreateNewTableRecords.ts

```diff
@@ -1,33 +1,94 @@
 import { useSelectedTableCellEditMode } from '@/object-record/record-table/record-table-cell/hooks/useSelectedTableCellEditMode';
+import { recordTablePendingRecordIdByGroupComponentFamilyState } from '@/object-record/record-table/states/recordTablePendingRecordIdByGroupComponentFamilyState';
+import { recordTablePendingRecordIdComponentState } from '@/object-record/record-table/states/recordTablePendingRecordIdComponentState';
+import { getDropdownFocusIdForRecordField } from '@/object-record/utils/getDropdownFocusIdForRecordField';
+import { useSetActiveDropdownFocusIdAndMemorizePrevious } from '@/ui/layout/dropdown/hooks/useSetFocusedDropdownIdAndMemorizePrevious';
 import { useSetHotkeyScope } from '@/ui/utilities/hotkey/hooks/useSetHotkeyScope';
-import { useContext } from 'react';
+import { useRecoilComponentCallbackStateV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentCallbackStateV2';
+import { useSetRecoilComponentStateV2 } from '@/ui/utilities/state/component-state/hooks/useSetRecoilComponentStateV2';
+import { useRecoilCallback } from 'recoil';
 import { v4 } from 'uuid';
+import { isDefined } from '~/utils/isDefined';
 
-export const useCreateNewTableRecord = (recordTableIdFromProps?: string) => {
-  const { recordTableId } = useContext(RecordTableContext);
-
-  const recordTableIdToUse = recordTableIdFromProps ?? recordTableId;
+export const useCreateNewTableRecord = (recordTableId: string) => {
+  const { objectMetadataItem } = useRecordIndexRootPropsContext();
 
   const { setSelectedTableCellEditMode } = useSelectedTableCellEditMode({
-    scopeId: recordTableIdToUse,
+    scopeId: recordTableId,
   });
 
   const setHotkeyScope = useSetHotkeyScope();
 
-  const { setPendingRecordId } = useRecordTable({
-    recordTableId: recordTableIdToUse,
-  });
+  const setPendingRecordId = useSetRecoilComponentStateV2(
+    recordTablePendingRecordIdComponentState,
+    recordTableId,
+  );
+
+  const recordTablePendingRecordIdByGroupFamilyState =
+    useRecoilComponentCallbackStateV2(
+      recordTablePendingRecordIdByGroupComponentFamilyState,
+      recordTableId,
+    );
+
+  const { setActiveDropdownFocusIdAndMemorizePrevious } =
+    useSetActiveDropdownFocusIdAndMemorizePrevious();
 
   const createNewTableRecord = () => {
-    setPendingRecordId(v4());
+    const recordId = v4();
+
+    setPendingRecordId(recordId);
     setSelectedTableCellEditMode(-1, 0);
     setHotkeyScope(DEFAULT_CELL_SCOPE.scope, DEFAULT_CELL_SCOPE.customScopes);
+
+    if (isDefined(objectMetadataItem.labelIdentifierFieldMetadataId)) {
+      setActiveDropdownFocusIdAndMemorizePrevious(
+        getDropdownFocusIdForRecordField(
+          recordId,
+          objectMetadataItem.labelIdentifierFieldMetadataId,
+          'table-cell',
+        ),
+      );
+    }
   };
 
+  const createNewTableRecordInGroup = useRecoilCallback(
```

Comment:
It could be split into to hooks ? : 
- useCreateNewTableRecordWithoutGroup
- useCreateNewTableRecordInGroup

Thus making everything clearer and forcing us to split the flow higher in the tree and keep to zones of code with group and without groups.

The most important thing to keep in mind : we always have more average developers than 100x developers working on a codebase, they should be able to understand simple code, so it's always better to have more code but keeping it simple (KISS) than having complex and compact code, even if it involves a bit of duplication.